src/intrinsics/ecma262/Object.js

```diff
@@ -265,8 +266,41 @@ export default function(realm: Realm): NativeFunctionValue {
     // 3. Let desc be ? obj.[[GetOwnProperty]](key).
     let desc = obj.$GetOwnProperty(key);
 
+    let getterFunc = desc && desc.get;
+    // If we are returning a descriptor with a NativeFunctionValue
+    // and it has no intrinsic name, then we create a temporal as this
+    // can only be done at runtime
+    if (
+      getterFunc instanceof NativeFunctionValue &&
+      getterFunc.intrinsicName === undefined &&
+      realm.useAbstractInterpretation
+    ) {
+      invariant(P instanceof Value);
+      // this will create a property descriptor at runtime
+      let result = AbstractValue.createTemporalFromBuildFunction(
+        realm,
+        ObjectValue,
+        [getOwnPropertyDescriptor, obj, P],
+        ([methodNode, objNode, keyNode]) => t.callExpression(methodNode, [objNode, keyNode])
+      );
```

Comment:
What's the difference between `createFromBuildFunction` and its temporal variant?

Reply:
We need to create a "temporal" entry in the generator here.

Reply:
A temporal value is something that needs to get created/computed at a particular point in the time line and may have further mutations that occur later on in the time line. An a-temporal value can get computed at any point and does not change.